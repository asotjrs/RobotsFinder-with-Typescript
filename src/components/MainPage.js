import React, {Component} from 'react';
import './MainPage.css';
import Cardlist from "../components/Cardlist";
import Scroller from "../components/Scroller";
import ErrorBoundry from "../containers/ErrorBoundry";
import Header from "../containers/Header";




class MainPage extends Component{



    render (){
        const robots=this.props.searchField===''? this.props.robots:this.props.searchField;

        return <div className="App tc">
            <Header onSearchChange={this.props.onSearchChange}/>
            {
                robots.trim()===""? <h1>Type Robots names Separated by "," Commas </h1>:
                    <Scroller><ErrorBoundry> <Cardlist robots={robots}/></ErrorBoundry> </Scroller>

            }
        </div>
    }
}

export default MainPage;
