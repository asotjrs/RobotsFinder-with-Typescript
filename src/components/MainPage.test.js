import React from "react";
import MainPage from "./MainPage";
import {shallow} from 'enzyme';

it('should test only mainPage', function () {

    const mockProperties={

        searchField:'',
        robots:"",
        isPending:false,
        onSearchChange:jest.fn()

    };

    expect(shallow(<MainPage {...mockProperties} />)).toMatchSnapshot();

});

it('should is_pending should be false', function () {
    const mockProperties2={

        searchField:'',
        robots:"",
        isPending:false,
        onSearchChange:jest.fn()

    };

    expect(shallow(<MainPage {...mockProperties2} />).instance().props.isPending).toEqual(false);


});