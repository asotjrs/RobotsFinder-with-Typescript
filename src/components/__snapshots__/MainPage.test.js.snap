// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test only mainPage 1`] = `
<div
  className="App tc"
>
  <Header
    onSearchChange={[MockFunction]}
  />
  <h1>
    Type Robots names Separated by "," Commas 
  </h1>
</div>
`;
